web/tests/test_discount.py

```
@@ -0,0 +1,93 @@
+        """
+        url = reverse("apply_discount_via_referrer") + f"?course_id={self.course.id}"
+        response = self.client.get(url, HTTP_REFERER="https://twitter.com")
+        # Expect a redirect to profile.
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, reverse("profile"))
+
+        # Discount record should exist.
+        discount = Discount.objects.filter(user=self.user, course=self.course, used=False).first()
+        self.assertIsNotNone(discount)
+
+    def test_discount_appears_in_profile(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_discount_appears_in_profile`

Add return type annotation: `None`

(ANN201)